dev/source-release.sh

```diff
@@ -87,11 +87,29 @@ tagrc="${tag}-rc${rc}"
 
 echo "Preparing source for $tagrc"
 
+# Generate git properties file as git info will not be as accessible in svn during final release
+# This SHA will technically be one behind the final release as we need to commit the file
+#echo "Generating git properties file for tarball"
+# $projectdir/./gradlew generateGitProperties --no-daemon
+# mv $projectdir/build/resources/main/iceberg-version.properties $projectdir/iceberg-version.properties
+
 echo "Adding version.txt and tagging release..."
 echo $version > $projectdir/version.txt
+# git add $projectdir/version.txt $projectdir/iceberg-version.properties
+# git commit -m "Add version.txt for release $version" $projectdir/version.txt $projectdir/iceberg-version.properties
 git add $projectdir/version.txt
 git commit -m "Add version.txt for release $version" $projectdir/version.txt
 
+# Regenerate the git properties file so its up to date, then move it out of build/ (which is in the .gitattributes
+# file and thus will be ignored from the archive). The file will be added to the release tarball, without committing,
+# so it has the correct SHA (from the previous step).
+#
+# TODO - Add a second task that generates _only_ the root properties file with just a subset of the info (so .dirty
+#        and user emails and hostnames don't appear or consider a virtual file in the git archive via --add-virtual-file).
+$projectdir/./gradlew generateGitProperties --no-daemon
+mv $projectdir/build/resources/main/iceberg-version.properties $projectdir/iceberg-version.properties
+git_props_file=$projectdir/iceberg-version.properties
+
```

Comment:
Please remove these unrelated changes. This PR should only contain the changes needed to add the `iceberg-build.properties` file.

Reply:
These are needed to get the file into the tarball so that it can be added in `deploy.gradle`. So I'm admittedly still experimenting with the process so that the output is correct.

Will revert these changes and we can add in another PR.

Reply:
Most of this stuff got pushed when running the `dev/source-release.sh` script to test this out.